### PR TITLE
refactor(core): Expression extensions shortlist (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
Adjustments to expression extensions listed [here](https://www.notion.so/n8n/Expression-extensions-shortlist-699e92b1b6694063bcaff71a346f5aaf#ede818eb1fb94bbf8f095a32f96978c4).

### CODE REVIEW

File: packages/workflow/src/Extensions/DateExtensions.ts [hunk 1]
```diff
@@ -117,6 +117,10 @@ function extract(inputDate: Date | DateTime, extraArgs: DatePart[]): number | Da
 		return Math.floor(diff / (1000 * 60 * 60 * 24));
 	}
 
+	if (part === 'week') {
+		part = 'weekNumber';
+	}
```
Comment (valya) [inline packages/workflow/src/Extensions/DateExtensions.ts:122]: I'd add this to `DATETIMEUNIT_MAP` instead.

File: packages/workflow/src/Extensions/ExpressionExtension.ts [hunk 2]
```diff
@@ -11,14 +11,6 @@ import { objectExtensions } from './ObjectExtensions';
 
 const EXPRESSION_EXTENDER = 'extend';
 
-function isBlank(value: unknown) {
-	return value === null || value === undefined || !value;
-}
-
-function isPresent(value: unknown) {
-	return !isBlank(value);
-}
```
Comment (valya) [inline packages/workflow/src/Extensions/ExpressionExtension.ts:20]: Removing these will result in `undefined` and `null` not working with `isEmpty`. If this is OK then they're fine to remove but thought I'd point it out if this wasn't intended.

**Review by valya — CHANGES_REQUESTED** [review]

File: packages/workflow/src/Extensions/ExpressionExtension.ts (hunk 2 above)
Comment (ivov) [reply packages/workflow/src/Extensions/ExpressionExtension.ts:20]: @nik8n ^Can you confirm?

File: packages/workflow/src/Extensions/DateExtensions.ts (hunk 1 above)
Comment (ivov) [reply packages/workflow/src/Extensions/DateExtensions.ts:122]: Added but then reverted because this works for `extract` but breaks `startOf`.

**Review by valya — DISMISSED** [review]

File: packages/workflow/src/Extensions/ExpressionExtension.ts (hunk 2 above)
Comment (ivov) [reply packages/workflow/src/Extensions/ExpressionExtension.ts]: Discussed and restored. @valya When you can please take another look.

File: packages/workflow/src/Extensions/ExpressionExtension.ts [hunk 3]
```diff
@@ -223,6 +238,12 @@ export function extend(input: unknown, functionName: string, args: unknown[]) {
 			// eslint-disable-next-line
 			return inputAny[functionName](...args);
 		}
+
+		console.log('genericExtensions', genericExtensions);
+		console.log('functionName', functionName);
```
Comment (nik8n) [inline packages/workflow/src/Extensions/ExpressionExtension.ts:243]: @ivov should these be in?

**Review by valya — APPROVED** [review]

**Comment by ivov** [comment]
Change of plans, had to keep making changes so [merged it](https://github.com/n8n-io/n8n/pull/5130/commits/55c33c470e3ea55cbf2ef1ef0ac0aaf357f40e03) directly into completions branch.

**Pull request closed without merging**